test/extended/dns/dns_libraries.go

```diff
@@ -0,0 +1,95 @@
+package dns
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+
+	kapiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	watchtools "k8s.io/client-go/tools/watch"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+// checkForPodLogFailures goes through the pod logs and determines if there was a failure
+// by looking for "fail" keyword in the logs.
+func checkForPodLogFailures(f *e2e.Framework, pod *kapiv1.Pod) {
+	By("submitting the pod to kubernetes")
+	podClient := f.ClientSet.CoreV1().Pods(f.Namespace.Name)
+	defer func() {
+		By("deleting the pod")
+		defer GinkgoRecover()
+		podClient.Delete(context.Background(), pod.Name, *metav1.NewDeleteOptions(0))
```

Comment:
I think the framework handles cleanup for you.  You use `f := e2e.NewDefaultFramework("dns-libraries")` at the start of the test, which creates a new namespace for the test, and then the framework deletes the namespace (and thus everything inside the namespace) after it runs.